Feature: 클러스터링 방법 변경

yoonjiseok opened the pull request:
### 📚 이슈 번호 <!-- 이슈 번호를 작성해주세요 ex) #11 -->

- close #이슈 번호 및 내용
#23 
### ✔️ 작업 내용

- 클러스터링 방법 변경 (하이퍼파리미터 선택 및 Jaccard similarity 사용)
- Neo4j 연결 (Docker 로 neo4j 실행하여 연결)

### 📝 리뷰 요청사항

- 

### 💻 테스트 결과
<img width="1823" height="118" alt="image" src="https://github.com/user-attachments/assets/7116de44-071a-438e-8516-bbb703504201" />
글의 구조적인 (structural) 가중치를 늘렸을 때 silhouette score 가 가장 높았습니다. 
아마 대부분의 마약 홍보글이 비슷한 템플릿으로 글을 써놨서 그랬다고 생각합니다.
-

Commit:
[FEATURE23] 임베딩 모델 변경

Commit:
[FEATURE] 클러스터링 방법 변경 기존 : 의미기반 방식 -> 변경 : jaccard 거리를 이용한 구조적 클러스터링 + 의미기반

Commit:
[FEATURE]임베딩 방식 변경(Jaccard)

Commit:
[FEATURE] clustering 모듈 작업

Commit:
[CHROE] Ignore tests directory

Fragarian commented on clustering/channel.py:
이 파일이 과거의 유산인 것 같긴 한데...일단 이전 프로젝트에서도 channel similarity는 사용하지 않고 있었습니다. 리팩토링이 완료되면 git 커밋 로그에서도 아예 삭제할 예정이라, 감안해주시면 좋겠습니다.

yoonjiseok replied:
넵

Fragarian commented on clustering/channel_come_in.py:
이것 역시 channel.py와 같은 이유로 삭제될 예정입니다.

yoonjiseok replied:
넵

Fragarian commented on pyproject.toml:
만약 pytest라는 모듈을 개발 과정에서만 사용하고 싶다면, uv add pytest --dev 명령어를 사용해서 개발 전용 패키지로 따로 빼낼 수도 있습니다. 한번 고려해보세요.

yoonjiseok replied:
네 확인했습니다

Fragarian committed:
Refactor: MongoDB 업데이트 로직 개선 및 데이터 직렬화 메서드 추가

## 주요 변경 사항

### 데이터 직렬화 메서드 추가
- **`core/mongo/post.py`**
  - `model_dump_only_insert` 메서드 추가:
    - `similarities` 필드를 제외한 나머지 데이터를 반환.
    - MongoDB의 `$setOnInsert` 연산에 활용.
  - `model_dump_only_update` 메서드 추가:
    - `similarities` 필드만 반환하도록 구현.
    - MongoDB의 `$set` 연산에 활용.

### 업데이트 로직 개선
- **`upsert_to_mongo` 메서드 수정**
  - 기존 `update_one` 호출 시 직렬화된 전체 데이터를 `$setOnInsert`에 전달했던 부분 수정.
  - `$setOnInsert`와 `$set` 구문으로 로직 분리:
    - `$setOnInsert`: 새로 삽입 시 사용할 데이터 (`model_dump_only_insert` 활용).
    - `$set`: 기존 데이터 업데이트 시 변경되는 필드 (`model_dump_only_update` 활용).

Fragarian committed:
Merge remote-tracking branch 'origin/feature/23' into feature/23

Fragarian reviewed (changes requested):
1. 변경된 posts 컬렉션의 필드명을 반영해서, 저장하는 필드명이나 projection map 등을 수정해 주세요. 원래의 구 데이터베이스(`retriever-release`)에 있던 게시글 데이터는 제가 `retriever-v2`로 필드명을 바꿔서 옮겨놓겠습니다.
2. 마찬가지로, 이제는 posts 컬렉션에 similarities라는 필드로 유사도까지 같이 저장합니다. 해당 내용을 반영해주세요. `core/mongo/post.py`에 있는 데이터 모델을 참조하면 됩니다.

Commit:
[FEATURE] 클러스터링 방법 변경

Commit:
[FEATURE] 다중 뷰 클러스터링 진행

Fragarian reviewed (approved):
수고하셨습니다.

Fragarian committed:
Merge branch 'develop' into feature/23
